src/result.ts

```diff
@@ -303,6 +343,10 @@ export class Ok<T, E> implements IResult<T, E> {
     return f(this.value)
   }
 
+  asyncAndThrough<F>(f: (t: T) => ResultAsync<unknown, F>): ResultAsync<T, F> {
+    return f(this.value).map<T>((_value: unknown) => this.value)
+  }
```

Comment:
```suggestion
  asyncAndThrough<R extends ResultAsync<unknown, unknown>>(
    f: (t: T) => R,
  ): ResultAsync<T, E | InferAsyncErrTypes<R>>
  asyncAndThrough<F>(f: (t: T) => ResultAsync<unknown, F>): ResultAsync<T, E | F>
  // eslint-disable-next-line @typescript-eslint/no-explicit-any, @typescript-eslint/explicit-module-boundary-types
  asyncAndThrough(f: (t: T) => ResultAsync<unknown, unknown>): any {
    return f(this.value).map(() => this.value)
  }
```
The signature should be overloaded and the error type should be union like `andThrough`.
If so, you can remove this comment out
https://github.com/supermacro/neverthrow/pull/467/files?w=1#diff-5e6df76f1ff5c2ddfbb0bee2e515236794778a429792a91d3b853bfecc884c16R471-R485